community/security/src/main/java/org/neo4j/server/security/auth/BasicAuthManager.java

```diff
@@ -130,14 +156,14 @@ public void setPassword( AuthSubject authSubject, String username, String passwo
         {
             throw new AuthorizationViolationException( "Invalid attempt to change the password for user " + username );
         }
-        if ( setPassword( username, password ) == null )
+        if ( setUserPassword( username, password ) == null )
         {
             throw new IllegalArgumentException( "User " + username + " does not exist" );
         }
     }
 
     @Override
-    public User setPassword( String username, String password ) throws IOException
+    public User setUserPassword( String username, String password ) throws IOException
```

Comment:
To me it reads badly with a `setX` method that returns the user. Can we think of a better name? And document the interface properly specifying that it can return `null` and what `null` means in this context.

To me it looks like we only use (except from tests) in constructs like 

```
if ( foo.setUserPassword("bar", "baz") == null )
```

wouldn't it make more sense to have it be a 

```
public boolean setUserPassword(....)
``` ?